### ISSUE TITLE
Fix false positives when block contains only nested child rules in block-no-empty

### ISSUE THREAD

**Issue by jarodtaylor** [stylelint/stylelint#3559]
> Clearly describe the bug

There are false positives for `block-no-empty` when the block only contains styles for nested elements.

> Which rule, if any, is the bug related to?

e.g. `block-no-empty`

> What CSS is needed to reproduce the bug?

```css
.foo {
  & .foo__child {
    color: #000000;
  }
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "block-no-empty": "true"
  }
}
```

> Which version of stylelint are you using?
`9.2.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Using stylelint-webpack-plugin

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to `postcss-nesting` nested properties.

> What did you expect to happen?

I expected it to fail, because `.foo` had no direct rules applied to it, so it would end up having an empty ruleset for it.

> What actually happened (e.g. what warnings or errors did you get)?

N/A

**Comment by jeddy3** [comment]
@jarodtaylor Thanks for the report and for using the template.

I think the rule is behaving as expected; neither of those blocks are, _within the source_, empty.

> so it would end up having an empty ruleset for it.

`postcss-nesting` is smart enough to strip the empty block. You can try it here: https://preset-env.cssdb.org/playground

> I expected it to fail, because .foo had no direct rules applied to it, so it would end up having an empty ruleset for it.

It sounds like you're looking to enforce the following style:

```css
.foo .foo__child {
  color: #000000;
}
```

over:

```css
.foo {
  & .foo__child {
    color: #000000;
  }
}
```

when a parent contains no declarations of its own.

There is no rule to do that currently, but you can create a [plugin](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/plugins.md) to do this.

**Comment by jeddy3** [comment]
Triage... closing as plugin candidate.
